tests/test_data_iterator.py

```diff
@@ -0,0 +1,116 @@
+from re import L
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+import pandas.testing as pd_test
+import pytest
+
+from packages.python.accern_data import create_data_client, DT_FORMAT
+from packages.python.accern_data.util import (
+    DEFAULT_CHUNK_SIZE,
+    EXAMPLE_URL,
+    load_json,
+)
+
+
+@pytest.mark.parametrize("chunk_size", [1, 5, 1000, None])
+def test_csv_full_iterator(chunk_size: Optional[int]) -> None:
+    start_date = "2022-01-03"
+    end_date = "2022-03-04"
+    client = create_data_client(EXAMPLE_URL, "SomeRandomToken")
+    client.set_mode("csv", split_dates=False, chunk_size=chunk_size)
+    dataframe = pd.read_csv("tests/data/data-2022.csv")
+    if chunk_size is not None:
+        n_full_chunks = dataframe.shape[0] // chunk_size
+    df_lengths = []
+    dfs: List[pd.DataFrame] = []
+    for df in client.iterate_range(start_date=start_date, end_date=end_date):
+        df_lengths.append(df.shape[0])
+        dfs.append(df)
+    concat_df = pd.concat(dfs)
+    assert (~concat_df.duplicated()).all(), "Duplicate entry is present."
+    if chunk_size is not None:
+        for idx in range(n_full_chunks):
+            assert df_lengths[idx] == chunk_size
+        if dataframe.shape[0] % chunk_size != 0:
+            assert df_lengths[-1] == dataframe.shape[0] % chunk_size
+    else:
+        for df in dfs:
+            assert df.shape[0] <= DEFAULT_CHUNK_SIZE
```

Comment:
don't rely on inequality -- you know exactly what the chunks are supposed to be. use those numbers